clarifai/client/model.py

```diff
@@ -47,10 +57,243 @@ def __init__(self,
+  def train(self, yaml_file: str = None) -> str:
+    """Trains the model based on the given yaml file or model params.
+
+    Args:
+        yaml_file (str): The yaml file for the model params.
+
+    Returns:
+        model_version_id (str): The model version ID for the model.
+
+    Example:
+        >>> from clarifai.client.model import Model
+        >>> model = Model(model_id='model_id', user_id='user_id', app_id='app_id')
+        >>> model_params = model.get_params(template='template', yaml_file='model_params.yaml')
+        >>> model.train('model_params.yaml')
+    """
+    if self.model_info.model_type_id not in TRAINABLE_MODEL_TYPES:
+      raise UserError(f"Model type {self.model_info.model_type_id} is not trainable")
+    if not yaml_file and len(self.training_params) == 0:
+      raise UserError("Provide yaml file or run 'model.get_params()'")
+
+    if yaml_file:
+      with open(yaml_file, 'r') as file:
+        params_dict = yaml.safe_load(file)
+    else:
+      params_dict = self.training_params
+
+    train_dict = params_parser(params_dict)
+    request = service_pb2.PostModelVersionsRequest(
+        user_app_id=self.user_app_id,
+        model_id=self.id,
+        model_versions=[resources_pb2.ModelVersion(**train_dict)])
+    response = self._grpc_request(self.STUB.PostModelVersions, request)
+    if response.status.code != status_code_pb2.SUCCESS:
+      raise Exception(response.status)
+    self.logger.info("\nModel Training Started\n%s", response.status)
+
+    return response.model.model_version.id
+
+  def training_status(self, version_id: str, training_logs: bool = False) -> Dict[str, str]:
+    """Returns the training status for the model version.
+
+    Args:
+        version_id (str): The version ID to get the training status for.
+        training_logs (bool): Whether to save the training logs in a file.
+
+    Returns:
+        training_status (Dict): Dictionary of training status for the model version.
+
+    Example:
+        >>> from clarifai.client.model import Model
+        >>> model = Model(model_id='model_id', user_id='user_id', app_id='app_id')
+        >>> model.training_status(version_id='version_id',training_logs=True)
+    """
+    if self.model_info.model_type_id not in TRAINABLE_MODEL_TYPES:
+      raise UserError(f"Model type {self.model_info.model_type_id} is not trainable")
+
+    request = service_pb2.GetModelVersionRequest(
+        user_app_id=self.user_app_id, model_id=self.id, version_id=version_id)
+    response = self._grpc_request(self.STUB.GetModelVersion, request)
+    if response.status.code != status_code_pb2.SUCCESS:
+      raise Exception(response.status)
+
+    if training_logs:
+      try:
+        if response.model_version.train_log:
+          log_response = requests.get(response.model_version.train_log)
+          log_response.raise_for_status()  # Check for any HTTP errors
+          with open(version_id + '.log', 'wb') as file:
+            file.write(log_response.content)
```

Comment:
```suggestion
            for chunk in log_response.iter_content(chunk_size=4096):  # 4KB
                  file.write(chunk)
```
scalable way to save large logs that may not fit into memory

Reply:
Sure